test/expressions/expression_test_helpers.hpp

```diff
@@ -268,17 +268,21 @@ auto bad_multiple_evaluations(const T& a) {
   return a + a;
 }
 
+/**
+ * The bad_* functions are meant to be used when generating a failure
+ */
 template <typename T>
 auto bad_wrong_value(const T& a) {
   if (std::is_same<T, plain_type_t<T>>::value) {
     return a(0, 0);
   }
+
   return a(0, 0) + 1;
 }
 
 template <typename T>
 auto bad_wrong_derivatives(const T& a) {
-  operands_and_partials<T> ops(a);
+  operands_and_partials<plain_type_t<T>> ops(a);
```

Comment:
The compiler was complaining here when `T` was an expression type because `operands_and_partials` is explicitly specialized for `Eigen::Matrix` for the fvar edge. I think this may have been a bug where we were calling the default `edge` implementation instead of the impl specialized for each autodiff type

Reply:
Hmm, I think we really want to support constructing `operands_and_partials` from expressions (at least ones that can be assigned to). So you should change this to `arena_t` and if that does not work it really is a bug that needs to be fixed. We are constructing `operands_and_partials` from arena types in distributions.

Reply:
This is still unresolved.

Reply:
Is a good point changed this to `ref_type_t`